### ISSUE TITLE
How to validate network?

### ISSUE THREAD

**Issue by willsilvano** [jmschrei/pomegranate#318]
Hello friends!

I have a project where I need to make predictions about school evasion.

I'm using the pomegrante api to generate the predictions.

My question is whether api currently has algorithms to validate the result.

Currently I train the network with 70% of the total dataset and the test with the other 30%. 

I go through the list of students in the test dataset and check if the result for evasion prediction was greater than 50%, and I classify as true evasion . I make the comparison with the target variable "evasion" that I already knew beforehand and in the end it generates an average of rights and wrongs.

Is there any other way I can do it through pomegranate?

Thank you!

**Comment by jmschrei** [comment]
Howdy. I'm not entirely sure what you mean. If you use `predict` instead of `predict_proba` it should give you hard classifications, which it seems like you're using. If you can share a snippet of the code I can provide a better answer. In general, pomegranate doesn't currently have any metrics supported. I leave that mostly to `sklearn.metrics`.

**Comment by willsilvano** [comment]
@jmschrei Thanks for answer!

Let me explain better.


I have a set of data on school dropout in which the target variable is called "**evasion**" that has the `True` and `False` states.

I load this dataset through the pandas dataframe, pack the rows to be randomly, and then split into two new dataframes: one with 70% of the data for network training and the other with 30% to test the network .

To generate the network I use `BayesianNetwork.from_samples` passing as parameter the training dataset and` chow-liu` algorithm.

After this, I iterate the test dataset, and in each iteration I do the prediction through the `predict_proba` method passing as parameter the student data.

As I already know in advance whether the student has evaded or not (`true` or `false`), I check the prediction for the variable "**evasion**" and which state has the highest value (`true `or `false`).

So I store in a list of right and wrong results, and at the end of the iteration of the test dataset I calculate an **average of the correct and wrong results**.

I got a mean of 90% correct cases, but I'd like to know if this method I used for validating the result is valid. Or if there is any other way to do it through the pomegrante.

Thank you again.

**Comment by jmschrei** [comment]
This sounds like how I would do it. You might want to use `predict` instead of `predict_proba` because it'll give you the maximally likely result directly instead of having to do it yourself. pomegranate doesn't have anything built in to do this for you though.
